python/cudf/cudf/tests/test_json.py

```diff
@@ -580,17 +587,18 @@ def test_default_float_bitwidth(default_float_bitwidth):
     assert df["b"].dtype == np.dtype(f"f{default_float_bitwidth//8}")
 
 
-def test_json_nested_basic(tmpdir):
-    fname = tmpdir.mkdir("gdf_json").join("tmp_json_nested_basic")
+def test_json_nested_basic():
+    bytes_obj = BytesIO()
     data = {
         "c1": [{"f1": "sf11", "f2": "sf21"}, {"f1": "sf12", "f2": "sf22"}],
         "c2": [["l11", "l21"], ["l12", "l22"]],
     }
     pdf = pd.DataFrame(data)
-    pdf.to_json(fname, orient="records")
+    pdf.to_json(bytes_obj, orient="records")
 
-    df = cudf.read_json(fname, engine="cudf_experimental", orient="records")
-    pdf = pd.read_json(fname, orient="records")
+    df = cudf.read_json(bytes_obj, engine="cudf", orient="records")
```

Comment:
same here, can we remove the parameter wherever the default value is set?
```suggestion
    df = cudf.read_json(bytes_obj, orient="records")
```

Reply:
Unfortunately, we can't. This is because the default for `lines` is `False` and our new json reader doesn't seem to support few `lines=False` cases for which when `engine='auto'` we toggle to `pandas` when `lines=False`:
```python
if engine == "auto":
        engine = "cudf" if lines else "pandas"
```

So to avoid switching to pandas reader in the above scenario, we will have explicitly pass `engine='cudf'` when `lines=False`. We still have support for `lines=False` in the new JSON reader but today we don't seem to have complete support.

Reply:
Is there an issue that can be opened / tracked / tagged here?

Reply:
@karthikeyann would know

Reply:
Would be nice to remark on the default engine selection in the PR description, I completely missed that this PR impacts only JSON Lines.

Reply:
One of the case our reader can't support:
[lines_false.json.txt](https://github.com/rapidsai/cudf/files/10470954/lines_false.json.txt)

Reply:
Isn't this a different orient than what we support?
After 23.02 work is done, we need to figure out how the orient parameter fits into the new parser. For one, it would allow us to avoid falling back to pandas for supported orients, even with lines=false.

Reply:
> Would be nice to remark on the default engine selection in the PR description, I completely missed that this PR impacts only JSON Lines.

Done 👍 Added a note in docstring too. We emit a warning in our code but never had it mentioned clearly in our docs.